app/assets/javascripts/index/directions-route-output.js

```diff
@@ -0,0 +1,140 @@
+    } else if (m < 1500) {
+      return OSM.i18n.t("javascripts.directions.distance_m", { distance: String(Math.round(m / 100) * 100) });
+    } else if (m < 5000) {
+      return OSM.i18n.t("javascripts.directions.distance_km", { distance: String(Math.round(m / 100) / 10) });
+    } else {
+      return OSM.i18n.t("javascripts.directions.distance_km", { distance: String(Math.round(m / 1000)) });
+    }
+  }
+
+  function formatHeight(m) {
+    return OSM.i18n.t("javascripts.directions.distance_m", { distance: Math.round(m) });
+  }
+
+  function formatTime(s) {
+    let m = Math.round(s / 60);
+    const h = Math.floor(m / 60);
+    m -= h * 60;
+    return h + ":" + (m < 10 ? "0" : "") + m;
+  }
+
+  const routeOutput = {};
+
+  routeOutput.write = function (content, engine, route) {
+    polyline
+      .setLatLngs(route.line)
+      .addTo(map);
+
+    const distanceText = $("<p>").append(
+      OSM.i18n.t("javascripts.directions.distance") + ": " + formatTotalDistance(route.distance) + ". " +
+      OSM.i18n.t("javascripts.directions.time") + ": " + formatTime(route.time) + ".");
+    if (typeof route.ascend !== "undefined" && typeof route.descend !== "undefined") {
+      distanceText.append(
+        $("<br>"),
+        OSM.i18n.t("javascripts.directions.ascend") + ": " + formatHeight(route.ascend) + ". " +
+        OSM.i18n.t("javascripts.directions.descend") + ": " + formatHeight(route.descend) + ".");
+    }
+
+    const turnByTurnTable = $("<table class='table table-hover table-sm mb-3'>")
+      .append($("<tbody>"));
+
+    content
+      .empty()
+      .append(
+        distanceText,
+        turnByTurnTable
+      );
+
+    for (const [i, [direction, instruction, dist, lineseg]] of route.steps.entries()) {
+      const row = $("<tr class='turn'/>").appendTo(turnByTurnTable);
+
+      if (direction) {
+        row.append("<td class='border-0'><svg width='20' height='20' class='d-block'><use href='#routing-sprite-" + direction + "' /></svg></td>");
+      } else {
+        row.append("<td class='border-0'>");
+      }
+      row.append(`<td><b>${i + 1}.</b> ${instruction}`);
+      row.append("<td class='distance text-body-secondary text-end'>" + formatStepDistance(dist));
+
+      row.on("click", function () {
+        popup
+          .setLatLng(lineseg[0])
+          .setContent(`<p><b>${i + 1}.</b> ${instruction}</p>`)
+          .openOn(map);
+      });
+
+      row
+        .on("mouseenter", function () {
+          highlight
+            .setLatLngs(lineseg)
+            .addTo(map);
+        })
+        .on("mouseleave", function () {
+          map.removeLayer(highlight);
+        });
+    }
+
+    const blob = new Blob([JSON.stringify(polyline.toGeoJSON())], { type: "application/json" });
+    URL.revokeObjectURL(downloadURL);
+    downloadURL = URL.createObjectURL(blob);
+
+    $("#directions_content").append(`<p class="text-center"><a href="${downloadURL}" download="${
```

Comment:
Shouldn't this, and the one that follows, be appending to the `content` parameter rather than looking it up in the DOM again?

Reply:
Yes, it should. Updated.